src/autolabel/few_shot/label_selector.py

```diff
@@ -46,23 +47,22 @@ def __init__(
         # Get the embeddings of the labels
         if self.label_descriptions is not None:
             (labels, descriptions) = zip(*self.label_descriptions.items())
-            embeddings = self.vectorStore._get_embeddings(descriptions)
-            for i, label in enumerate(labels):
-                self.labels_embeddings[label] = embeddings[i]
+            self.labels = list(labels)
+            self.labels_embeddings = torch.Tensor(
+                self.vectorStore._get_embeddings(descriptions)
+            )
         else:
-            embeddings = self.vectorStore._get_embeddings(self.labels)
-            for i, label in enumerate(labels):
-                self.labels_embeddings[label] = embeddings[i]
+            self.labels_embeddings = torch.Tensor(
+                self.vectorStore._get_embeddings(self.labels)
+            )
+        print(type(self.labels_embeddings))
 
     def select_labels(self, input: str) -> List[str]:
         """Select which labels to use based on the similarity to input"""
-        input_embedding = self.vectorStore._get_embeddings([input])
-
-        scores = []
-        for label, embedding in self.labels_embeddings.items():
-            similarity = cos_sim(embedding, input_embedding)
-            # insert into scores, while maintaining sorted order
-            bisect.insort(scores, (similarity, label))
+        input_embedding = torch.Tensor(self.vectorStore._get_embeddings([input]))
+        scores = cos_sim(input_embedding, self.labels_embeddings).view(-1)
+        scores = list(zip(scores, self.labels))
+        scores.sort(key=lambda x: x[0])
 
         # remove labels with similarity score less than self.threshold*topScore
         return [
```

Comment:
I think the list should be filtered first based on the threshold and then we should select scores[-self.k: ]. This is so that we retrieve k items if the number of labels above threshold is > k